org.eclipse.scout.rt.rest/src/main/java/org/eclipse/scout/rt/rest/id/IdSignatureClientRequestFilter.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (c) 2010, 2024 BSI Business Systems Integration AG
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.scout.rt.rest.id;
+
+import java.io.IOException;
+
+import jakarta.annotation.Priority;
+import jakarta.ws.rs.Priorities;
+import jakarta.ws.rs.client.ClientRequestContext;
+import jakarta.ws.rs.client.ClientRequestFilter;
+
+import org.eclipse.scout.rt.dataobject.id.IId;
+import org.eclipse.scout.rt.dataobject.id.IdCodec.IdCodecFlag;
+import org.eclipse.scout.rt.platform.Bean;
+
+/**
+ * This {@link ClientRequestFilter} adds the {@link #ID_SIGNATURE_HTTP_HEADER} to the {@link ClientRequestContext},
+ * which will be used later on to determine if {@link IId}s need to be signed (see {@link IdCodecFlag#SIGNATURE} for
+ * more details).
+ */
```

Comment:
maybe add a sentence about "this fliter is not auto-registered", use org.eclipse.scout.rt.rest.client.AbstractRestClientHelper#getRequestFiltersToRegister in your rest client helper implementation to register this filter"